### PR TITLE
[core] Unify testing of the value related props

### PR DESCRIPTION
**Pull request by flaviendelangle**
## Work

- [x] Add `describeValidation` on `MonthCalendar` 
- [x] New `describeValue` that checks the `value` / `defaultValue` behaviors for all components and the `onClose` / `onAccept` / `onChange` behaviors for all pickers (when using the various action bar actions for instance)

## Side notes

- We have missing `describeConformance` on pickers, with the new file structure it's very easy to find the missing one, I'll add them in a follow up PR and try to reduce the `skip` that no longer apply
- Adapt `describeValue` for range components later

### CODE REVIEW

File: docs/pages/x/api/date-pickers/desktop-next-time-picker.json [hunk 1]
```diff
@@ -92,10 +92,7 @@
   },
   "name": "DesktopNextTimePicker",
   "styles": { "classes": [], "globalClasses": {}, "name": "MuiDesktopNextTimePicker" },
```
Comment (flaviendelangle) [inline docs/pages/x/api/date-pickers/desktop-next-time-picker.json:94]: We need to improve the `parseTest` method on the core repo to handle the new test file names
I propose to do it in a follow up PR, as in any case it was already broken for most components

File: packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts [hunk 2]
```diff
@@ -249,7 +249,7 @@ export const usePickerValue = <
   const [rawValue, setValue] = useControlled({
     controlled: inValue,
     default: defaultValue ?? valueManager.emptyValue,
-    name: 'usePickerState2',
+    name: 'usePickerValue',
```
Comment (flaviendelangle) [inline packages/x-date-pickers/src/internals/hooks/usePicker/usePickerValue.ts:252]: Not related

**Review by alexfauquette — COMMENTED** [review]
Having one `describe.xxx.test.tsx` file per component sound very clear 👍

File: docs/pages/x/api/date-pickers/desktop-next-time-picker.json (hunk 1 above)
Comment (flaviendelangle) [reply docs/pages/x/api/date-pickers/desktop-next-time-picker.json:94]: The information lost in this file are parsed from the `describeConformance` test
And `parseTest` look from a test file in the same folder as the component file

**Review by alexfauquette — APPROVED** [review]

File: test/utils/pickers-utils.tsx [hunk 3]
```diff
@@ -178,13 +186,30 @@ export const getClockMouseEvent = (
...
+  }
+
+  let itemCount: number;
+  if (view === 'minutes') {
+    itemCount = 60;
+  } else if (view === '12hours') {
+    itemCount = 12;
+  } else {
+    itemCount = 24;
+  }
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply test/utils/pickers-utils.tsx:202]: I agree that `itemCount = 24` made no sense, that's fixed
I'll let the error for now, don't have the time to test if its works (and we don't have tests using it)

You can have a look if you want :+1:

**Review by LukasTy — APPROVED** [review]

File: docs/pages/x/api/date-pickers/desktop-next-time-picker.json (hunk 1 above)
Comment (oliviertassinari) [reply docs/pages/x/api/date-pickers/desktop-next-time-picker.json:94]: #7294 is continuing this discussion.

**Pull request merged**